src/server/pfs/server/source.go

```diff
@@ -37,7 +37,7 @@ func NewSource(commitInfo *pfs.CommitInfo, fs fileset.FileSet, opts ...SourceOpt
 	}
 	s := &source{
 		commitInfo: commitInfo,
-		fileSet:    fileset.NewDirInserter(fs, sc.prefix),
+		fileSet:    fileset.NewDirInserter(fileset.NewPathValidator(fs), sc.prefix),
```

Comment:
I think the wrapping of the fileset with a validator should just be the first line in this function, rather than here. I think it is more maintainable that way, and it communicates intention a bit better imo (i.e. always ensure the passed in fileset goes through a validator).